run_app.sh

```diff
@@ -0,0 +1,17 @@
+TEMPDIR=`mktemp -d`
+
+if [ -z "$INVENTORY_SVC_URL" ];
+  then echo "INVENTORY_SVC_URL is not set" && exit 1;
+fi
+
+if [ -z "$LOG_LEVEL" ];
+  then LOG_LEVEL='info'
+fi
+
+if [ -z "$NUM_WORKERS" ];
+  then NUM_WORKERS=2
+fi
+
+prometheus_multiproc_dir=$TEMPDIR gunicorn wsgi -w $NUM_WORKERS -b 0.0.0.0:8080 --log-level=$LOG_LEVEL --access-logfile=-
```

Comment:
What owns or controls the `prometheus_multiproc_dir` environment variable?

Reply:
It works differently than what we discussed on IRC. `prometheus_multiproc_dir` is controlled by the bash script, which is also in charge of making sure the temp dir exists and is cleaned up afterwards.

If we wanted to have the same stats across pods, we would need a shared volume for this dir. However I think for this PR it's OK to assume single-pod.

Reply:
Controlled by the bash script I get, but *what* code uses it? One of the libraries? And has to be set before the app initializes?

Reply:
ah I see what you mean. The metrics display code has:

```python
def metrics():
    registry = CollectorRegistry()
    multiprocess.MultiProcessCollector(registry)
    prometheus_data = generate_latest(registry)
    return prometheus_data
```

The `multiprocess.MultiProcessCollector(registry)` line initializes a collector that requires the env var. If it's not set, then the app will raise an error at startup.

Reply:
Is it possible to set that either here https://github.com/RedHatInsights/drift-backend/pull/32/files#diff-3f41e546893dc64b71aaacad12cad815R3 or before the `application.run()`

Reply:
I agree its kind of a strange place to initialize a registry. It's in the example but that might be just to make the example more simple. I'll see what it's doing exactly and if it can be moved so it's not initialized over and over

Reply:
I thought I was going to have to write a singleton (ugh), but it looks like I can throw the registry on the app context maybe. Trying that now.

Reply:
@ehelms registry has been put onto the app context so it's not initalized over and over